runtime/astar/src/lib.rs

```diff
@@ -844,6 +844,29 @@ impl pallet_xc_asset_config::Config for Runtime {
     type WeightInfo = pallet_xc_asset_config::weights::SubstrateWeight<Self>;
 }
 
+#[derive(Default)]
+pub struct DispatchPrecompileFilter;
+
+impl InstanceFilter<RuntimeCall> for DispatchPrecompileFilter {
+    fn filter(&self, c: &RuntimeCall) -> bool {
+        match c {
+            RuntimeCall::Utility(pallet_utility::Call::batch { calls }) => {
+                for call in calls {
+                    if !DispatchPrecompileFilter::default().filter(call) {
+                        return false;
+                    }
+                }
```

Comment:
> I don't think this is needed - I didn't know this until recently but even when a call is made inside the batch, the same filter will be applied again.
 
I couldn't find any code for the same, can you kindly share the code?

In the integration test I am just importing the struct which has implemented the trait `InstanceFilter` and testing the calls using this filter.

Reply:
I checked how the preocompile dispatch is implemented, but it's not done how I expected. So current implementation is good. 👍 

I'd suggest to move this struct, for all runtimes, into the `precompile.rs` file since it doesn't need to be defined in `lib.rs`.